java/arcs/android/storage/database/DatabaseImpl.kt

```diff
@@ -705,6 +706,59 @@ class DatabaseImpl(
         }
     }
 
+    override suspend fun runGarbageCollection() {
+        val twoDaysAgo = JvmTime.currentTimeMillis - Duration.ofDays(2).toMillis()
+        writableDatabase.transaction {
+            val db = this
+            rawQuery(
+                """
+                    SELECT storage_key_id, storage_key, orphan, MAX(entity_refs.id) IS NULL AS noRef
+                    FROM entities
+                    LEFT JOIN storage_keys ON entities.storage_key_id = storage_keys.id
+                    LEFT JOIN entity_refs ON entity_storage_key = storage_keys.storage_key
+                    GROUP BY storage_key_id, storage_key, orphan
```

Comment:
I think the only thing you'll need to group-by is `entity_refs.id`, since that's what the `MAX` is operating on.  I'd double-check though...

Reply:
that doesn't work, entity_refs.id is null for all entities with no ref, so they would be all grouped